hazelcast/src/main/java/com/hazelcast/wan/impl/WanReplicationServiceImpl.java

```diff
@@ -268,4 +285,130 @@ public void removeWanEventCounters(String serviceName, String objectName) {
         // EE WAN replication not supported in OS
         return Collections.emptyList();
     }
+
+    @Override
+    public void beforeMigration(PartitionMigrationEvent event) {
+        for (DelegatingWanReplicationScheme wanReplication : wanReplications.values()) {
```

Comment:
seems we have streaming api candidates in this class?

Reply:
True, it does seem repetitive.